### PR TITLE
Update release checklist origanization & links

### PR DESCRIPTION
**Pull request by electricRGB**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_checklist_template.md [hunk 1]
```diff
@@ -6,25 +6,32 @@ labels: ''
...
 
 ---
-Soft freeze:   
-Hard freeze: 
+Soft freeze:
+Hard freeze:
 
-Branch setup, steps typically done when creating new version branch:
+## Soft Freeze Items
+### Branch setup, steps typically done when creating new version branch:
```
Comment (dnarcese) [inline .github/ISSUE_TEMPLATE/release_checklist_template.md:13]:
```suggestion
```
Probably don't need this

File: .github/ISSUE_TEMPLATE/release_checklist_template.md [hunk 2]
```diff
@@ -6,25 +6,32 @@ labels: ''
...
 - [ ] Update Version Number in code (Eng task, use `update_version.sh`)
 - [ ] Update `Client/Info.plist` MozWhatsNewTopic (Eng task)
 - [ ] Create version specific Sentry project
 - [ ] Update SENTRY_DSN secret in Bitrise (per release)
-- [ ] Add Bitrise trigger for release branch 
----
+- [ ] Add Bitrise trigger for release branch
+
+### Soft Freeze Optional
+These items should be completed as soon as possible, preferrably on soft code freeze day. Due to timing issues, they may not be available until hard code freeze, and must
```
Comment (dnarcese) [inline .github/ISSUE_TEMPLATE/release_checklist_template.md:21]: Incomplete sentence

File: .github/ISSUE_TEMPLATE/release_checklist_template.md [hunk 3]
```diff
@@ -6,25 +6,32 @@ labels: ''
...
+- [ ] Check for [security advisories](https://github.com/mozilla-mobile/firefox-ios/wiki/Release-Build-Checklist/#security-advisories)
+- [ ] File P.I. request
+
+## Hard Code Freeze
+
 - [ ] Tag release in GitHub (Eng task)
     - [ ] Link to commit diff between versions
     - [ ] Add volunteer contributions
-- [ ] File P.I. request
 - [ ] Release Notes updated
```
Comment (dnarcese) [inline .github/ISSUE_TEMPLATE/release_checklist_template.md:25]: Let's make this ordering
- rel notes
- screenshots and locales
- submit
- rollout

**Review by dnarcese — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release_checklist_template.md [hunk 4]
```diff
@@ -6,25 +6,32 @@ labels: ''
...
+
+### Soft Freeze Optional
+These items should be completed as soon as possible, preferrably on soft code freeze day. Due to timing issues, they may not be available until hard code freeze, and must
+
 - [ ] Ensure string export was completed for L10N (if needed)
-- [ ] Check for [security advisories](https://github.com/mozilla-mobile/firefox-ios/wiki/Release-Build-Checklist/#security-advisories) 
+- [ ] Check for [security advisories](https://github.com/mozilla-mobile/firefox-ios/wiki/Release-Build-Checklist/#security-advisories)
+- [ ] File P.I. request
+
+## Hard Code Freeze
```
Comment (dnarcese) [inline .github/ISSUE_TEMPLATE/release_checklist_template.md:27]:
```suggestion
## Hard Freeze Items
```
Just for consistency

**Review by dnarcese — CHANGES_REQUESTED** [review]

**Comment by electricRGB** [comment]
@dnarcese comments have been addressed. :) Also.... I have no idea why I left that sentence incomplete. lol

**Review by dnarcese — APPROVED** [review]

**Review by andikaleonardo — APPROVED** [review]

**Pull request merged**